doc/introduction/remote.rst

```diff
@@ -228,104 +154,122 @@ array([[0, 0, 1, 0, 1, 0, 1, 0],
-         0.19903677-0.43076659j, -0.50320649-0.44750373j, -0.01617065-0.52755812j, -0.19729219+0.06200712j
+Finally, an incomplete job can be *cancelled* by calling :meth:`job.cancel() <~.Job.cancel>`.
 
-    # Initial states are two-mode squeezed states
-    S2gate(1.0, 0.0) | [0, 4]
-    S2gate(1.0, 0.0) | [1, 5]
-    S2gate(1.0, 0.0) | [2, 6]
-    S2gate(1.0, 0.0) | [3, 7]
 
-    # Apply the unitary matrix above to
-    # the first pair of modes, as well
-    # as a beamsplitter
-    Interferometer(U) | [0, 1, 2, 3]
-    BSgate(0.543, -0.123) | [0, 1]
+Hardware compilation
+--------------------
 
-    # Duplicate the above unitary for
-    # the second pair of modes
-    Interferometer(U) | [4, 5, 6, 7]
-    BSgate(0.543, -0.123) | [4, 5]
+When creating a quantum program to run on hardware, Strawberry Fields can compile
+any collection of the following gates into a multi-mode unitary:
 
-    # Perform a PNR measurement in the Fock basis
-    MeasureFock() | [0, 1, 2, 3, 4, 5, 6, 7]
+* `General beamsplitters <https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.BSgate.html>`_ (:class:`~.ops.BSgate`),
 
+* `Mach-Zehnder interferometers <https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.MZgate.html>`_ (:class:`~.ops.MZgate`), or
 
-**Note:** You may use :func:`~.random_interferometer` to generate arbitrary random unitaries.
+* `rotations/phase shifts <https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.Rgate.html>`_ (:class:`~.ops.Rgate`).
 
-This program will execute following the same steps as above; :class:`~.RemoteEngine` will automatically
-compile the program to match the layout of the chip.
+Furthermore, several automatic decompositions are supported:
 
-You may wish to view the compiled program; this can be easily done in Python using
-the ``Program.print`` method:
+* You can use the :class:`~.ops.Interferometer` command to directly pass a
+  unitary matrix to be decomposed and compiled to match the device architecture.
+  This performs a rectangular decomposition using Mach-Zehnder interferometers.
 
+* You can use :class:`~.ops.BipartiteGraphEmbed` to embed a bipartite graph on
+  the GBS chip. Note, however, that the decomposed squeezing values depends on the graph
+  structure, so only bipartite graphs that result in equal squeezing on all
+  modes can be executed on currently available chips.
 
->>> from strawberryfields import RemoteEngine
->>> from strawberryfields.io import load
->>> prog = load("test.xbb")
->>> prog = prog.compile("chip2")
->>> prog.print()
+Before sending the program to the cloud platform to be executed, however, Strawberry Fields
+must **compile** the program to match the physical architecture or layout of the photonic chip.
+This happens implicitly when using the remote engine, however we can use the :meth:`~.Program.compile`
```

Comment:
Does the engine recompile already-compiled programs?

Reply:
This is the `eng.run` logic:
```python
# if the program hasn't been compiled for this backend, do it now
target = self.backend.circuit_spec
if target is not None and p.target != target:
    p = p.compile(target, **compile_options)
```

So compilation only occurs if the existing program target does not match the engine backend.